### ISSUE TITLE
varianttype and alternateBases and matename hould use oneof construct in OpenAPI specs

### ISSUE THREAD

**Issue by blankdots** [ga4gh-beacon/specification#278]
Related to https://github.com/ga4gh-beacon/specification/issues/273 and https://github.com/ga4gh-beacon/specification/issues/277#issuecomment-483760720

Seems like:
* `alternateBases` or `variantType` cannot be used with `mateName`;
* if a query specifies `variantType=BND` it seems like it is a looks `mateName` should not be present.

Or am i missing something?

**Comment by mbaudis** [comment]
You need mateName=chro... + variantType=BND to specify a fusion event with known partners.

**Comment by blankdots** [comment]
@mbaudis it does not seem like that from the API description, maybe worth detailing this particular thing in a shape of form.
But just to be clear if one would specify `mateName` shouldn't `variantType` be assumed as `BND` by default, or are there any cases we are missing?

**Comment by mbaudis** [comment]
@blankdots You’re correct; but IMO a well-formed fusion query should specify BND, even if it is implicit.
I’m not sure how to go further about this - specification or documentation for query types - on a technical level; a “oneOf” construct should in principle apply to query prototypes, not single attributes.  @sdelatorrep?

**Comment by blankdots** [comment]
ok, open to any suggestion, how to specify `oneof` was a suggestion. 
Also for an implementation standpoint it would be another check that needs to be performed that when `mateName` is used with `variantType` to be sure that any service querying the API used the correct value: `BND` - and that is ok, just highlighting that this might be useful when clarifying in the API specs.
